File: dmoj/ccc08j4.js
// https://dmoj.ca/problem/ccc08j4
// https://dmoj.ca/submission/2621387
// should be run with v8dmoj

var line;
var postfix;
var i;
var a;
var b;
while(true){
    line = gets()
    if(line === '0'){
        break;
    }
    line = line.split(' ');
    line.reverse();
    postfix = [];
    for(i=0;i<line.length;i++){
        if(line[i] === '+' || line[i] === '-'){
            a = postfix.pop();
            b = postfix.pop();
            postfix.push(a+" "+b+" "+line[i]);
        }
        else{
            postfix.push(line[i]);
        }
    }
    print(postfix.join(" "));
}

File: dmoj/ccc08j4.py
# https://dmoj.ca/problem/ccc08j4
# https://dmoj.ca/submission/2609708

import sys
while True:
    line = sys.stdin.readline()[:-1].split()
    if line == ['0']:
        break
    postfix = []
    line = line[::-1]
    for i in line:
        if i == '+' or i == '-':
            a = postfix.pop()
            b = postfix.pop()
            postfix.append(a+' '+b+' '+i)
        else:
            postfix.append(i)
    print(' '.join(postfix))